gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/TimeAggregationsWithGroupByRequestHandler.java

```diff
@@ -1,31 +1,54 @@
 package org.hypertrace.gateway.service.explore;
 
-import com.google.common.collect.ImmutableSet;
-import java.util.Set;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Streams;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.hypertrace.core.attribute.service.v1.AttributeMetadata;
 import org.hypertrace.gateway.service.common.AttributeMetadataProvider;
+import org.hypertrace.gateway.service.common.util.AttributeMetadataUtil;
 import org.hypertrace.gateway.service.common.util.ExpressionReader;
 import org.hypertrace.gateway.service.common.util.QueryServiceClient;
 import org.hypertrace.gateway.service.v1.common.Expression;
 import org.hypertrace.gateway.service.v1.common.Filter;
 import org.hypertrace.gateway.service.v1.common.LiteralConstant;
 import org.hypertrace.gateway.service.v1.common.Operator;
+import org.hypertrace.gateway.service.v1.common.Row;
+import org.hypertrace.gateway.service.v1.common.TimeAggregation;
 import org.hypertrace.gateway.service.v1.common.Value;
 import org.hypertrace.gateway.service.v1.common.ValueType;
 import org.hypertrace.gateway.service.v1.explore.ExploreRequest;
 import org.hypertrace.gateway.service.v1.explore.ExploreResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class TimeAggregationsWithGroupByRequestHandler implements IRequestHandler {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(TimeAggregationsWithGroupByRequestHandler.class);
 
+  private final AttributeMetadataProvider attributeMetadataProvider;
   private final RequestHandler normalRequestHandler;
   private final TimeAggregationsRequestHandler timeAggregationsRequestHandler;
 
   TimeAggregationsWithGroupByRequestHandler(
       QueryServiceClient queryServiceClient, AttributeMetadataProvider attributeMetadataProvider) {
+    this.attributeMetadataProvider = attributeMetadataProvider;
     this.normalRequestHandler = new RequestHandler(queryServiceClient, attributeMetadataProvider);
     this.timeAggregationsRequestHandler =
         new TimeAggregationsRequestHandler(queryServiceClient, attributeMetadataProvider);
   }
 
+  @VisibleForTesting
```

Comment:
😢 (this version is actually better, but I'd rather remove the old constructor and make the actual code use this one too)